src/DocumentDB.ChangeFeedProcessor/PartitionManagement/ILeaseEx.cs

```diff
@@ -0,0 +1,18 @@
+﻿// ----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+// ----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement
+{
+    /// <summary>
+    /// Provides extensibility for <see cref="ILease"/> interface.
+    /// <see cref="ILease"/> implementation can also can implement this interface to utilize the ability to set lease acquire reason from <see cref="IParitionLoadBalancingStrategy"/>.
+    /// </summary>
+    public interface ILeaseEx
```

Comment:
OK but I think it would be better to have more specific interface name just for this case, e.g. ILeaseWithAcquireReason or something like that. Can you discuss this with @kadukf and get to agreement?

Reply:
What do you think about ILeaseAcquireReasonProvider? This interface does not inherit ILease thus ILeaseWithAcquireReason looks a bit misleading to me

Reply:
At little long but OK